specs/0121-translation-memory-management.md

```diff
@@ -0,0 +1,43 @@
+- Feature Name: Translation Memory Management
+- Created: 2024-08-28
+- Associated Issue: #3031
+
+# Summary
+
+Add ability to edit, delete and upload Translation Memory (TM) entries.
+
+# Motivation
+
+Translation memory is a fundamental tool for faster and more consistent translations, and has become even more important with pretranslation, since it's used to train custom machine translation engines in Google AutoML Translation.
+
+Support for TM in Pontoon is limited:
+* Localizers cannot edit or delete existing TM entries directly.
+* TM entries can be changed indirectly via translation workspace, but only if the corresponding strings are still available for translation.
+* Localizers cannot upload their own translation memories.
+
+# Feature explanation
+
+A new "Translation Memory" tab is added to the Team pages, which lists all TM entries for the team. For each entry, source string and translation are provided, clicking on which leads to the translation workspace for the corresponding strings (using the list `query` parameter). Entries are listed in alphabetic order and only shown once for each unique combination of source string and translation.
```

Comment:
Going back to visibility, I'm concerned about resources and DDoS here (given we just experienced one), also considering the infinite scrolling UX.

Would it make sense to make it visible only to people signed in?

Reply:
Listing TM entries is much faster than querying TM for fuzzy matches (which is available to all users). 

I have the feature as presented in the initial version of the spec implemented locally and it's even (and quite a bit) faster than the Projects tab (which is also public), probably because there aren't any JOINs.

That said, I expect the view to get slower when we start grouping by unique combinations of source strings and translations. I wouldn't want to speculate in the spec though whether making the view private will be needed to mitigate that.